tests/Tokenizer/TokensTest.php

```diff
@@ -29,6 +29,587 @@
+                1,
+                '<?php
+  '.'
+',
+                '<?php
+;
+  '.'
+',
+            ],
+            // --- before token
+            'a' => [
+                3,
+                '<?php # 0',
+                '<?php # 0
+;',
+            ],
+            'b' => [
+                3,
+                '<?php
+/* 1 */
+  ',
+                '<?php
+/* 1 */
+;
+  ',
+            ],
+            'c' => [
+                3,
+                '<?php
+/* 1 */
+  # 1.2',
+                '<?php
+/* 1 */
+;
+  # 1.2',
+            ],
+            'd' => [
+                4,
+                '<?php
+                /* 2 */  '.'
+                ',
+                '<?php
+                /* 2 */  ;
+                ',
+            ],
+            'e' => [
+                3,
+                '<?php # 3
+    ',
+                '<?php # 3
+    '.'
+                ;',
+            ],
+            '4/' => [
+                2,
+                '<?php            ',
+                '<?php            ;',
+            ],
+            '5/' => [
+                1,
+                '<?php ',
+                "<?php\n;",
+            ],
+            // --- before and after
+            [
+                3,
+                '<?php
+# A7
+# B7
+',
+                '<?php
+# A7
+;
+# B7
+',
+            ],
+            [
+                4,
+                '<?php
+                # A9
+                # B
+                ',
+                '<?php
+                # A9
+                ;        '.'
+                # B
+                ',
+            ],
+            // --- after token
+            '10/' => [
+                1,
+                '<?php ',
+                '<?php
+;     ',
+            ],
+            '11/' => [
+                2,
+                '<?php
+',
+                '<?php
+
+;     ',
+            ],
+            '12/' => [
+                1,
+                '<?php ',
+                '<?php
+;',
+            ],
+
+            '14/' => [
+                3,
+                '<?php
+# a',
+                '<?php
+# a
+;     ',
+            ],
+            '16/' => [
+                1,
+                '<?php
+ # 1
+',
+                '<?php
+; # 1
+',
+            ],
+            '17/a' => [
+                3,
+                '<?php
+# 17a
+ # 18a
+',
+                '<?php
+# 17a
+; # 18a
+',
+            ],
+            '17/b' => [
+                3,
+                '<?php
+# 17b
+  # 18b
+',
+                '<?php
+# 17b
+ ; # 18b
+',
+            ],
+            '17/c' => [
+                3,
+                '<?php
+# 17c
+ # 18c
+',
+                '<?php
+# 17c
+ ;# 18c
+',
+            ],
+            '17/d' => [
+                2,
+                '<?php
+/* 17d*/# 18d
+',
+                '<?php
+/* 17d*/;# 18d
+',
+            ],
+            // --- special cases
+            '18/' => [
+                0,
+                '',
+                '<?php ',
+            ],
+            '20/' => [
```

Comment:
Are these indices somehow useful?

Reply:
When a case fails it more easy to find back which one it is.

Reply:
I removed the confusing indices